src/checked_transaction.rs

```diff
@@ -22,46 +43,50 @@ use core::borrow::Borrow;
 /// If you need to modify an inner state, you need to get inner values
 /// (via the `Into<(Tx, Tx ::Metadata)>` trait), modify them and check again.
 ///
-/// Each ascending checked status includes all previous checks. So `Signature` means that
-/// `Stateless` also is passed.
-///
 /// # Dev note: Avoid serde serialization of this type. Since checked tx would need to be
 /// re-validated on deserialization anyways, it's cleaner to redo the tx check.
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
-pub enum Checked<Tx: IntoChecked> {
-    /// The transaction passed checks without the usage of the state of the blockchain.
-    ///
-    /// The rules defined in the specification:
-    /// https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#transaction
-    Stateless(Tx, Tx::Metadata),
-    /// Signatures in the transaction are valid.
-    Signatures(Tx, Tx::Metadata),
+pub struct Checked<Tx: IntoChecked> {
+    transaction: Tx,
+    metadata: Tx::Metadata,
+    checks_bitmask: Checks,
 }
 
 impl<Tx: IntoChecked> Checked<Tx> {
-    pub fn transaction(&self) -> &Tx {
-        match self {
-            Checked::Stateless(tx, _) => tx,
-            Checked::Signatures(tx, _) => tx,
+    pub fn new(transaction: Tx, metadata: Tx::Metadata, checks_bitmask: Checks) -> Self {
```

Comment:
hmm I don't think setting the checked status should be allowed by the public API, it should only be set internally as a result of the check being performed.

Reply:
It should be private, copy/paste error)